Ds3/Helpers/Jobs/Job.cs

```diff
@@ -65,6 +67,12 @@ public TSelf WithCancellationToken(CancellationToken cancellationToken)
             return (TSelf)(IBaseJob<TSelf, TItem>)this;
         }
 
+        public TSelf WithMetadata(IMetadataAccess metadataAccess)
+        {
+            this._metadataAccess = metadataAccess;
```

Comment:
We should add a check to make sure that if this is called after the Transfer method is called, an exception is thrown.